extensions/Push/lib/Connector/Phabricator.pm

```diff
@@ -55,12 +54,40 @@ sub send {
     my $logger = Bugzilla->push_ext->logger;
 
     my $data = $message->payload_decoded;
-    my $bug_data = $self->_get_bug_data($data) || return 0;
-    my $bug = Bugzilla::Bug->new( { id => $bug_data->{id}, cache => 1 } );
+    my $bug = $self->_get_bug_by_data($data) || return PUSH_RESULT_OK;
+
+    # Q: Should we check to see if a group we don't care about changed?
+    # If so, then this doesn't automatically mean we need to make req to Phab?
+    # ======================================================
+    $is_public = is_public($bug);
+
+    # Check to see if any synced groups have changed
+    # =================================================
+    my $phab_sync_groups = Bugzilla->params->{phabricator_sync_groups}
+        || ThrowUserError('invalid_phabricator_sync_groups');
+    my $sync_group_names = [ split('[,\s]+', $phab_sync_groups) ];
+
+    my $bug_groups = $bug->groups_in;
+    my $bug_group_names = [ map { $_->name } @$bug_groups ];
+
+    my @set_groups = intersect($bug_group_names, $sync_group_names);
+
+    # "If bug privacy groups do not have any matching synchronized groups,
+    # then leave revision private and it will have be dealt with manually."
+    # ==
+    # Is this: https://trello.com/c/9edMlHrl/399-2-when-locking-down-a-revision-because-of-unknown-security-groups-i-want-to-see-a-helpful-message-in-the-revision-comments-and-t
+    # ????
+    # Should we return PUSH_RESULT_OK; ?
+    # ========================================
+    if (!$is_public && !@set_groups) {
+        ThrowUserError('invalid_phabricator_sync_groups');
```

Comment:
Per discussion in IRC, this is the case where we need to set the revision to viewable by admins only. This part is easier in that we do not have to create the new custom policy. We can just do differential.revision.edit and set the view policy directly to 'admins'. After adding a comment we would return here.